File: seed/rust-model/trace/src/execution_session_state.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct ExecutionSessionState {
    #[serde(rename = "lastTimeContacted")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub last_time_contacted: Option<String>,
    #[serde(rename = "sessionId")]
    pub session_id: String,
    #[serde(rename = "isWarmInstance")]
    pub is_warm_instance: bool,
    #[serde(rename = "awsTaskId")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub aws_task_id: Option<String>,
    pub language: Language,
    pub status: ExecutionSessionStatus,
}

File: seed/ruby-sdk-v2/trace/lib/execution_session_state.rb
# frozen_string_literal: true

module Submission
    module Types
        class ExecutionSessionState < Internal::Types::Model
            field :last_time_contacted, Array, optional: true, nullable: true
            field :session_id, String, optional: true, nullable: true
            field :is_warm_instance, Boolean, optional: true, nullable: true
            field :aws_task_id, Array, optional: true, nullable: true
            field :language, Language, optional: true, nullable: true
            field :status, ExecutionSessionStatus, optional: true, nullable: true
        end
    end
end
